### PR TITLE
Image building and pushing for free5gc-operator

### PR DESCRIPTION
**Pull request by radoslawc**
Images built from every commit will have prow BUILD_ID as a version and those build from release tags v.X.Y.Z will have that tag name as version

### CODE REVIEW

File: prow/config/config.yaml [hunk 1]
```diff
@@ -80,6 +80,81 @@ postsubmits:
...
+    cluster: default
+    always_run: true
+    branches:
+    - "^v.*$"
+    annotations:
+      description: Build and Push Image to DockerHub when release tag is created
+    decorate: true
+    decoration_config:
+      censor_secrets: true
+    max_concurrency: 1
```
Comment (electrocucaracha) [inline prow/config/config.yaml:95]: Do we have to [restrict the number](https://github.com/kubernetes/test-infra/blob/master/prow/apis/prowjobs/v1/types.go#L174-L179) of jobs to run in parallel?
```suggestion
    max_concurrency: 0
```

File: prow/config/config.yaml [hunk 2]
```diff
@@ -80,6 +80,81 @@ postsubmits:
...
+    cluster: default
+    always_run: true
+    branches:
+    - "main"
+    annotations:
+      description: Build and Push Image to DockerHub every commit
+    decorate: true
+    decoration_config:
+      censor_secrets: true
+    max_concurrency: 1
```
Comment (electrocucaracha) [inline prow/config/config.yaml:131]: @radoslawc this can also run in parallel
```suggestion
    max_concurrency: 0
```

**Review by gvbalaji — APPROVED** [review]
/approve

File: prow/config/config.yaml [hunk 3]
```diff
@@ -80,6 +80,81 @@ postsubmits:
...
+        - |
+          executor --context=${PWD}/free5gc-operator \
+            --dockerfile=Dockerfile --destination=nephio/free5gc-operator:${PULL_BASE_REF}
+        volumeMounts:
+          - name: kaniko-secret
+            mountPath: /kaniko/.docker/
+        resources:
+          requests:
+            cpu: 2
+            memory: 2Gi
```
Comment (s3wong) [inline prow/config/config.yaml:112]: No a problem / blocker. But 2 vCPUs and 2 GB memory seems like a LOT for the operator...

**Pull request merged**